CodeGeneration/Sources/generate-swift-syntax/GenerateSwiftSyntax.swift

```diff
@@ -173,13 +174,23 @@ struct GenerateSwiftSyntax: ParsableCommand {
           verbose: verbose
         )
       } catch {
+        errorsLock.lock()
         errors.append(error)
+        errorsLock.unlock()
       }
     }
 
-    if let firstError = errors.first {
-      // TODO: It would be nice if we could emit all errors
-      throw firstError
+    if errors.count > 0 {
+      struct ComposititeError: Error, CustomStringConvertible {
+        var errors: [Error]
+        var description: String {
+          "Multiple errors emitted: \n" +
+          errors
+            .map(String.init(describing: ))
+            .joined(separator: "\n")
```

Comment:
Do they already have a newline, or would it be worth adding an extra newline here? Do you happen to have an example of the output here for multiple errors?

Reply:
I can make you some real output in a few hours, but in my original testing `NSError` does not include a newline, and that's the error that we're most likely to encounter in this code path today due to the use of `FileManager`. 

I think in general that it will depend on how the error implements its `description` or `debugDescription` var and that the safest option is to add a newline even if we can find some errors that happen to do this for us.

Reply:
Sorry, to be clear, we definitely want at least one newline here. My question was about whether we might want *two* so that there's a clearer cut between the errors.

Reply:
Ah, yeah that makes a lot more sense. Here's the sample output of an `NSError` in this context: 
```
Error: Multiple errors emitted: 
Error Domain=test error Code=0 "(null)"
Error Domain=test error Code=0 "(null)"
Error Domain=test error Code=0 "(null)"
Error Domain=test error Code=0 "(null)"
Error Domain=test error Code=0 "(null)"
Error Domain=test error Code=0 "(null)"
```
I think in an ideal world it would look something like 
```
Multiple errors emitted: {

    Error Domain=test error Code=0 "(null)"

    Error Domain=test error Code=0 "(null)"

   ...

}
```
because they're separate, but grouped. But that would require adding a means of indenting text (in case of multi-line errors) which might be overkill. But I'm happy to add the indentation and newline, or just the second newline, or leave it out, whichever you prefer.